content/toolsAndResources/nge.md

```diff
@@ -0,0 +1,126 @@
+
+Subnode is rarely to found on nodes and it refers to nodes that have exactly one other node they can be connected to. Mostly this is used to reduce complexity on a large node moving optional operations or parameters to another node. While they are not commonly found on nodes, seeing the subnode icon is a good indication to look for another node that can take this node as an input.
+
+Anytime a node requires a specific input to be wired, the console log will display an error saying the graph cannot be evaluated until all required inputs have been connected. As showin in the image below, the add node needs to have both the left and right inputs connected to complete the evaluation of the graph. Since only one input has a connection, there is an error displayed in the console describing which block is missing a connection.
+
+![Required connections are displayed in console log](/img/tools/nge/requiredConnection.jpg)
+
+### Shortcuts for Connecting Nodes
+
+There are several shortcuts to help speed up connecting nodes which can save a lot of mouse clicks. The first is to select a node and then press the spacebar to open up the node list shortcut. Select a node to add to the graph and the added node will attempt to wire a connection to the selected node if there is a compatible input type on the added node. This method won't always result in an automatically wired connection if the editor can't find a compatible input type. And if no automatic wire is desired, simply ensure there is no selected node by clicking on an empty area of the graph to deselect all nodes before adding a node with the node list shortcut. 
+
+![Nodes added from node list shortcut will attempt to wire to selected nodes](/img/tools/nge/spaceBarToWire.jpg)
+
+The other shortcut to be aware of is dropping a node onto an existing wire to automatically connect compatible inputs and outputs. The node needs to already be in the graph for this to work, so drag it from the node list to the graph first. Then when dragging the node over a wire, the wire will highlight if the dragged node has inputs that are compatible with the data type represented by the wire. When releasing the mouse button, the editor will attempt to connect both inputs and outputs of the new node to maintain the original flow for the graph. There are, however, instances where a new node dropped on a wire in this way will still prevent the graph from being evaluated, mostly due to the new node having requirements that are not met. For example, in the image below the `add` node is dropped on the wire connecting a float to the size input of the `Box` node. The add node will connect the first compatible inputs and outputs it finds to retain the original flow, but in this case the graph won't build because the add node needs another float type connection to the right input before it can evaluate.

```

Comment:
Do we want to mention having ALT+ move link?

Reply:
I added alt + move as well as alt  + move to an empty part of the graph to create a matching input node.

Reply:
in latest commit to PR